src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp

```diff
@@ -245,12 +245,24 @@ namespace OpenMS
             for (auto const& kv : filter_criteria.component_qcs[c_qc_it].meta_value_qc)
             {
               // std::cout << "MetaData" << std::endl; //debugging
-              if (!checkMetaValue(features[feature_it].getSubordinates()[sub_it], kv.first, kv.second.first, kv.second.second))
+              const Feature& component = features[feature_it].getSubordinates()[sub_it];
+              const String& meta_value_key = kv.first;
+              const double meta_value_l = kv.second.first;
+              const double meta_value_u = kv.second.second;
+              if (component.metaValueExists(meta_value_key))
               {
-                c_qc_pass = false;
-                c_qc_fail_message_vec.push_back("metaValue[" + kv.first + "]");
+                const double meta_value = (double)component.getMetaValue(meta_value_key);
+                if (!checkRange(meta_value, meta_value_l, meta_value_u))
+                {
+                  c_qc_pass = false;
+                  c_qc_fail_message_vec.push_back("metaValue[" + meta_value_key + "]");
+                }
+                ++c_tests_count;
```

Comment:
Much better.  It may be necessary to pass in a counter to the CheckMetaValue function to keep this same logic.